Refresh option not available for some resources that are not closed p…

DaveCarpeneto opened the pull request:
Refresh option not available for some resources that are not closed projects #2538

Change made since the fix to #1438 unintentionally removed the "refresh" contextual menu for resources that are not projects. 

With this change the "refresh" contextual menu is shown if ANY navigator selection is either (A) an open project, or (B) a non-project resource. Put another way: the 'refresh' item is NOT shown if ALL selections are closed projects.

Fixes https://github.com/eclipse-platform/eclipse.platform.ui/issues/2538

DaveCarpeneto committed:
Refresh option not available for some resources that are not closed projects #2538

Change made since the fix to #1438 unintentionally removed the "refresh" contextual menu for resources that are not projects. 

With this change the "refresh" contextual menu is shown if ANY navigator selection is either (A) an open project, or (B) a non-project resource. Put another way: the 'refresh' item is NOT shown if ALL selections are closed projects.

Fixes https://github.com/eclipse-platform/eclipse.platform.ui/issues/2538

eclipse-platform-bot committed:
Version bump(s) for 4.35 stream

DaveCarpeneto commented:
This replaces https://github.com/eclipse-platform/eclipse.platform.ui/pull/2539 

@jukzi : apologies for the second pull request; I'm new to GitHub. For reference, the changes requested were: 

 - please use for-each loop instead -> done
 - please add explanation why projects.size() < selection.size() is used here. That is not self-explaining / maybe introduce a variable like selectionContainsNonProject -> done
 - The commit message does not follow our best practice -> changed

jukzi commented:
> apologies for the second pull request;

no worries. The jenkins build failed with "An unexpected error occurred while launching the test runtime (process returned error code 13)". I do not know why i restarted the job. We need a green result before it could be merged.

jukzi commented on bundles/org.eclipse.ui.navigator.resources/src/org/eclipse/ui/internal/navigator/resources/actions/ResourceMgmtActionProvider.java:
```diff
@@ -139,19 +139,23 @@ public void fillContextMenu(IMenuManager menu) {
 				hasBuilder = false;
 			}
 		}
+
```
the change would look better if it would avoid unnecessary new lines added.

DaveCarpeneto replied:
Fixed in the commits added today

jukzi commented on bundles/org.eclipse.ui.navigator.resources/src/org/eclipse/ui/internal/navigator/resources/actions/ResourceMgmtActionProvider.java:
```diff
@@ -117,14 +116,15 @@ public void fillContextMenu(IMenuManager menu) {
+				// we've set all booleans of interest; no need to loop any further
+				break;
+			}
 			if (project == null) {
```
Can you explain why project can be null here and how that would affect "selectionContainsNonProject"?

DaveCarpeneto replied:
There's no way that this could be null, however given I don't know the code base well I was trying to touch as little as possible in my changes, so I left this as-is. 

From what I can see commit 18df0075dfd74d07ae9c384ede85e94233b36cdd from 2017 could have / should have removed this, but did not. Regardless: this is cleaned up in the commit added today

jukzi commented:
Does not look bad overall. Could you please try to add a junit test for it?

DaveCarpeneto committed:
Update ResourceMgmtActionProvider.java

DaveCarpeneto committed:
Update NavigatorTestSuite.java

Refresh option not available for some resources that are not closed projects https://github.com/eclipse-platform/eclipse.platform.ui/issues/2538

Change made since the fix to #1438 unintentionally removed the "refresh" contextual menu for resources that are not projects. 

With this change the "refresh" contextual menu is shown if ANY navigator selection is either (A) an open project, or (B) a non-project resource. Put another way: the 'refresh' item is NOT shown if ALL selections are closed projects.

Fixes #2538

DaveCarpeneto committed:
Create ResourceMgmtActionProviderTests.java

Refresh option not available for some resources that are not closed projects https://github.com/eclipse-platform/eclipse.platform.ui/issues/2538

Change made since the fix to #1438 unintentionally removed the "refresh" contextual menu for resources that are not projects. 

With this change the "refresh" contextual menu is shown if ANY navigator selection is either (A) an open project, or (B) a non-project resource. Put another way: the 'refresh' item is NOT shown if ALL selections are closed projects.

Fixes #2538

eclipse-platform-bot committed:
Version bump(s) for 4.35 stream

DaveCarpeneto commented:
> Does not look bad overall. Could you please try to add a junit test for it?

Added in today's commit. Please let me know if anything looks incorrect (I used other unit tests in this section for reference, but since I couldn't find a guide as to what is expected (other than https://github.com/eclipse-platform/.github/blob/main/CONTRIBUTING.md 's missive that tests should run and pass (which these do)).

DaveCarpeneto committed:
Merge branch 'eclipse-platform:master' into DaveCarpeneto-issue2538-1

jukzi commented on tests/org.eclipse.ui.tests.navigator/src/org/eclipse/ui/tests/navigator/resources/ResourceMgmtActionProviderTests.java:
```diff
@@ -0,0 +1,221 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
```
please stick to the format ` * Copyright (c) 2024 <your name or compan> and others.`

DaveCarpeneto replied:
I had hoped to avoid putting my name on this via the "more generic copyright header" option mentioned @ https://www.eclipse.org/projects/handbook/#ip-copyright-headers , but it's all good / change added.

jukzi replied:
Interesting i was not aware of that option. We have not used it in this project so far.
Quoting `Alternatively, project teams can adopt a more generic copyright header`  its not even clear how to get such a decision. Mailing list? Really per project? Maybe @eclipse-platform-project-leads can clarify

DaveCarpeneto committed:
Refresh option not available for some resources that are not closed projects 

Refresh option not available for some resources that are not closed projects #2538

Change made since the fix to #1438 unintentionally removed the "refresh" contextual menu for resources that are not projects.

With this change the "refresh" contextual menu is shown if ANY navigator selection is either (A) an open project, or (B) a non-project resource. Put another way: the 'refresh' item is NOT shown if ALL selections are closed projects.

Fixes #2538

DaveCarpeneto commented:
Hi @jukzi - anything else needed on this pull request from my end? I don't think I see anything, but I'm still very new to github :-)

jukzi commented:
This repo does not allow merge commits. Please squash all changes into a single commit and rebase on head. Once the automated checks are successful someone might take the time to test your change manually.

DaveCarpeneto commented:
> _Please squash all changes into a single commit_

To reduce mess / risk I've simply created a new pull request with a single commit https://github.com/eclipse-platform/eclipse.platform.ui/pull/2601
